UI/SidePanel.cpp

```diff
@@ -1058,11 +1063,45 @@ namespace {
             return false;
         if (IsAvailable(ship, system_id, empire_id) &&
             ship->HasTroops() &&
-            ship->OrderedInvadePlanet() == INVALID_OBJECT_ID )
+            ship->OrderedInvadePlanet() == INVALID_OBJECT_ID)
         { return true; }
         return false;
     };
 
+    bool AvailableToBombard(TemporaryPtr<const Ship> ship, int system_id, int empire_id) {
+        if (!ship)
+            return false;
+        TemporaryPtr<Fleet> fleet = GetFleet(ship->FleetID());
+        if (!fleet)
+            return false;
+        if (IsAvailable(ship, system_id, empire_id) &&
+            ship->CanBombard() &&
+            ship->OrderedBombardPlanet() == INVALID_OBJECT_ID)
+        { return true; }
+        return false;
+    };
+
+    /** Content tags that note if a Ship should be auto-selected for bombarding a Planet.
+     *  These tags are determined from the TAG_BOMBARD_PREFIX tags of @a ship and potentially match those of a Planet.
+     *  If the Ship contains the content tag defined in TAG_BOMBARD_ALWAYS, only that tag will be returned.
+     */
+    std::vector<std::string> BombardTagsForShip(TemporaryPtr<const Ship> ship) {
+        std::vector<std::string> retval;
+        if (!ship)
+            return retval;
+        std::set<std::string> tags = ship->Tags();
+        for (std::set<std::string>::iterator tag_it = tags.begin(); tag_it != tags.end(); ++tag_it) {
+            if (*tag_it == TAG_BOMBARD_ALWAYS) {
+                retval.clear();
+                retval.push_back(*tag_it);
+                break;
+            } else if ((tag_it->length() > TAG_BOMBARD_PREFIX.length()) &&
```

Comment:
probably OK, but could have used boost::starts_with
http://stackoverflow.com/questions/1878001/how-do-i-check-if-a-c-string-starts-with-a-certain-string-and-convert-a-sub